src/lib/pages/tx-details/components/TxInfo.tsx

```diff
@@ -0,0 +1,49 @@
+import type { FlexProps } from "@chakra-ui/react";
+import { Text, chakra, Flex } from "@chakra-ui/react";
+
+import { ExplorerLink } from "lib/components/ExplorerLink";
+import { LabelText } from "lib/components/LabelText";
+import type { TxData } from "lib/services/txService";
+import { formatInteger } from "lib/utils";
+
+interface TxInfoProps extends FlexProps {
+  txData: TxData;
+}
+
+const Container = chakra(Flex, {
+  baseStyle: {
+    flexDir: "column",
+    gap: 6,
+    w: "250px",
+  },
+});
+
+export const TxInfo = ({ txData, ...flexProps }: TxInfoProps) => (
+  <Container {...flexProps}>
+    <LabelText label="Network">{txData.chainId}</LabelText>
+    <LabelText label="Block">
+      <ExplorerLink
+        value={txData.height.toString()}
+        type="block_height"
+        canCopyWithHover
+      />
+    </LabelText>
+    <LabelText label="Transaction Fee">
+      {txData.formattedFee || (
```

Comment:
When the left hand is an optional value, I prefer to use '??' as a fallback value.